include/common/core/freertos_timer.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "FreeRTOS.h"
+#include "timers.h"
+
+namespace freertos_timer {
+
+// pdMS_TO_TICKS converts milliseconds to ticks. This can only be used for
+// FreeRTOS tick rates less than 1000 Hz.
+template <TickType_t timer_period = pdMS_TO_TICKS(2)>
+class FreeRTOSTimer {
+  public:
+    /*
+     * A software timer class. The priority of software timers in FreeRTOS is
+     * currently set to 6. Any tasks utilizing this timer should have either the
+     * same priority or higher priority than 6 for execution.
+     */
+    FreeRTOSTimer(const char* name, TimerCallbackFunction_t callback)
```

Comment:
you should c++ify this imo. take a `std::function` instead of a `TimerCallbackFunction_t`, make an `extern "c" void callback(void* me) { me->do_callback(); }` that you pass to `xTimerCreateStatic`. That way callers can drop in a lambda, a wrapper around a member function pointer, whatever they want.